### PR TITLE
Update SOE

### PR DESCRIPTION
**Pull request by Eluxor**
:cl: Eluxor
tweak: Reworked all SOE gear, berets, helmets and jumpsuits galore. "Dress your own SOE trooper" game not included. SOE armor is no longer spaceproof.
rscadd: SOE gets a very new spacesuit with a jetpack for breaching and capturing operations in space, due their new armor.
rscadd: Added two new guns that fit the newly made SOE style. The KV-32 Double barrelled custom shotgun, magazine !!FED!! with 4 rounds and the old Trusty weapon of space pirate captains... THE HANDGONNE! A 14.5mm pistol.
imageadd: Added new SoE gear sprites.
/:cl:

### CODE REVIEW

File: code/modules/halo/clothing/urf_commando.dm [hunk 1]
```diff
@@ -1,59 +1,114 @@
+#define ITEM_HAND 'code/modules/halo/clothing/head.dmi'
+#define URF_OVERRIDE 'code/modules/halo/clothing/urf_commando.dmi'
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm:2]: Both of these `#define`s need to be undefined at the end of the file

File: code/modules/halo/clothing/urc_commando_variants.dm [hunk 2]
```diff
@@ -1,94 +1,85 @@
...
-
-	item_state = "death_a_worn"
-	icon_state = "death_a_obj"
-
-/obj/item/clothing/head/helmet/urfc/death
-	name = "URFC Death Helmet"
-
-	item_state = "death_worn"
-	icon_state = "death_helmet"
+//MASKS
```
Comment (bloxgate) [inline code/modules/halo/clothing/urc_commando_variants.dm:82]: There doesn't seem to be any masks after here?

File: code/modules/halo/clothing/urf_commando.dm [hunk 3]
```diff
@@ -1,59 +1,114 @@
...
+	item_state = "commander_worn"
+	icon_state = "commander_helmet"
+	item_flags = THICKMATERIAL
+	body_parts_covered = HEAD
+	armor = list(melee = 20, bullet = 25, laser = 10,energy = 10, bomb = 0, bio = 20, rad = 5)
+	integrated_hud = /obj/item/clothing/glasses/hud/tactical
+
+/obj/item/clothing/head/helmet/soe
+	name = "SOE Space Helmet"
+	desc = "Non-Standard issue short-EVA capable helmet issued to (REDACTED) forces"
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: I don't really think you need to put the `(REDACTED)` in, but that's not anything major.

File: code/modules/halo/clothing/urf_commando.dm [hunk 4]
```diff
@@ -1,59 +1,114 @@
...
+	armor = list(melee = 55, bullet = 50, laser = 55, energy = 45, bomb = 40, bio = 50, rad = 25)
 	body_parts_covered = UPPER_TORSO | LOWER_TORSO | LEGS | FEET | ARMS | HANDS
+	armor_thickness = 20
+	item_flags = THICKMATERIAL
+	var/slots = 4
+	var/max_w_class = ITEM_SIZE_SMALL
+
+/obj/item/clothing/suit/armor/special/soe
+	name = "SOE Spacesuit"
+	desc = "Heavyweight,somewhat durable armour issued to (REDACTED) for increased survivability in the field."
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: `Heavyweight, somewhat`. And ditto the above on `(REDACTED)`

File: code/modules/halo/clothing/urf_commando.dm [hunk 5]
```diff
@@ -1,59 +1,114 @@
...
+	item_flags = STOPPRESSUREDAMAGE|THICKMATERIAL
+	body_parts_covered = UPPER_TORSO | LOWER_TORSO | ARMS | LEGS| FEET | HANDS
+	flags_inv = HIDEGLOVES|HIDESHOES|HIDEJUMPSUIT|HIDETAIL
 	cold_protection = UPPER_TORSO | LOWER_TORSO | LEGS | FEET | ARMS | HANDS
 	heat_protection = UPPER_TORSO | LOWER_TORSO | LEGS | FEET | ARMS | HANDS
 	min_cold_protection_temperature = SPACE_SUIT_MIN_COLD_PROTECTION_TEMPERATURE
 	max_heat_protection_temperature = FIRESUIT_MAX_HEAT_PROTECTION_TEMPERATURE
+	item_state_slots = list(slot_l_hand_str = "urf_armor", slot_r_hand_str = "urf_armor")
 	armor_thickness = 20
+	slowdown_general = -0.5
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: Negative `slowdown_general` makes you faster. Are you sure that's what you want to do.

File: code/modules/halo/clothing/urf_commando.dm [hunk 6]
```diff
@@ -109,4 +176,55 @@
...
+
+/obj/item/weapon/tank/jetpack/urfc
+	name = "SOE Jetpack (oxygen)"
+	desc = "It works well in a void."
+	icon = 'code/modules/halo/clothing/urf_commando.dmi'
+	icon_override = 'code/modules/halo/clothing/urf_commando.dmi'
+	icon_state = "soe_tank_obj"
+	item_state =  "soe_tank"
+
+/obj/item/weapon/tank/jetpack/urfc/New()
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: If you make this a sub-type of the existing oxygen jetpack, you won't need to have the extra `New()` proc here.

File: code/modules/halo/clothing/urf_commando.dm [hunk 7]
```diff
@@ -109,4 +176,55 @@
...
+	body_parts_covered = FACE|HEAD
+	item_flags = AIRTIGHT|FLEXIBLEMATERIAL
+	gas_transfer_coefficient = 0.01
+	permeability_coefficient = 0.01
+	siemens_coefficient = 0.9
+	item_flags = 0
+	pull_mask = 1
+	w_class = ITEM_SIZE_SMALL
+
+/obj/item/clothing/mask/gas/soebalaclavaopen
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: This should probably be a subtype of the above gas mask.

File: code/modules/halo/clothing/urf_commando.dm [hunk 8]
```diff
@@ -109,4 +176,55 @@
...
+	name = "SOE Open Balaclava"
+	icon = ITEM_HAND
+	icon_state = "merc_balaclava_open"
+	item_state = "merc_balaclava_open"
+	icon_override = URF_OVERRIDE
+	flags_inv = HIDEFACE
+	body_parts_covered = FACE
+	item_flags = AIRTIGHT|FLEXIBLEMATERIAL
+	gas_transfer_coefficient = 0.01
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: This is insanely airtight for something that's supposedly open.

File: code/modules/halo/weapons/pistols.dm [hunk 9]
```diff
@@ -66,3 +66,29 @@
 	else
 		icon_state = "SOCOM_unloaded"
 	. = ..()
+
+//Hand Cannon
+
+/obj/item/weapon/gun/projectile/heavysniper/handgonne
```
Comment (bloxgate) [inline code/modules/halo/weapons/pistols.dm:72]: This already exists in #1744. There's going to be problems if either of these are merged due to that.

**Review by bloxgate — CHANGES_REQUESTED** [review]
Just a few suggestions. Besides these, I think the changelog would read a little better if it was reworded a bit. Maybe make it so that the entries are a bit more self contained? I.e., the second entry doesn't require you to have read the first to know what the `But` is referring to?

File: code/modules/halo/clothing/urc_commando_variants.dm [hunk 10]
```diff
@@ -1,94 +1,92 @@
...
+	name = "SOE Officer Beret"
+	desc = "An officer beret. Weirdly made of some kind of bulletproof material."
+	item_state = "beretofficer_worn"
+	icon_state = "beretofficer_helmet"
 
-	item_state = "death_a_worn"
-	icon_state = "death_a_obj"
+/obj/item/clothing/head/helmet/urfccommander/beretmerc
+	name = "SOE Beret"
+	desc = "An red beret. Weirdly made of some kind of bulletproof material."
```
Comment (bloxgate) [inline code/modules/halo/clothing/urc_commando_variants.dm]: `An` -> `A`

File: code/modules/halo/clothing/urf_commando.dm [hunk 11]
```diff
@@ -1,59 +1,114 @@
...
 	item_state_slots = list(slot_l_hand_str = "urf_helmet", slot_r_hand_str = "urf_helmet")
+	item_flags = THICKMATERIAL
+	body_parts_covered = HEAD|FACE
+	flags_inv = BLOCKHAIR
+	armor = list(melee = 60, bullet = 35, laser = 25,energy = 25, bomb = 20, bio = 50, rad = 25)
+	integrated_hud = /obj/item/clothing/glasses/hud/tactical
+
+/obj/item/clothing/head/helmet/urfccommander
+	name = "SOE Commander Hat"
+	desc = "An commander hat. Weirdly made of some kind of bulletproof material."
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: A

File: code/modules/halo/clothing/urf_commando.dm [hunk 12]
```diff
@@ -109,4 +176,37 @@
...
 	slot_l_hand_str = "c_pack_med",
 	slot_r_hand_str = "c_pack_med",
-	)
\ No newline at end of file
+	)
+
+
+//JETPACK
+
+/obj/item/weapon/tank/jetpack/urfc
```
Comment (bloxgate) [inline code/modules/halo/clothing/urf_commando.dm]: Yeah... this isn't the change I asked you to make. To get rid of the `New` proc for this, it has to be a subtype of the existing oxygen jetpack. This is a subtype of the empty jetpack at the moment.

File: code/modules/halo/weapons/pistols.dm [hunk 13]
```diff
@@ -66,3 +66,29 @@
...
+	accuracy = 0
+	wielded_item_state = "handgonne"
+
+	item_icons = list(
+		slot_l_hand_str = 'code/modules/halo/weapons/icons/Weapon_Inhands_left.dmi',
+		slot_r_hand_str = 'code/modules/halo/weapons/icons/Weapon_Inhands_right.dmi',
+		)
+
+/obj/item/weapon/gun/projectile/heavysniper/handgonne/scope()
+	 return
```
Comment (bloxgate) [inline code/modules/halo/weapons/pistols.dm:94]: This proc does literally nothing. Are you sure you need to have it?

**Review by bloxgate — CHANGES_REQUESTED** [review]
Looking much better! Mainly just a few issues with the descriptions on a few items. I'm going to ask Cael to take a look over this since he's been handling a fair bit of the URF/SOE stuff lately.

File: code/modules/halo/weapons/pistols.dm (hunk 13 above)
Comment (Eluxor) [reply code/modules/halo/weapons/pistols.dm:94]: Yes, its due the handgonne being a subtype of the heavy sniper. I need that to remove the scope. As the heavy sniper has a scope and the handgonne doesnt.

File: code/modules/halo/clothing/urc_commando_variants.dm (hunk 10 above)
Comment (Eluxor) [reply code/modules/halo/clothing/urc_commando_variants.dm]: fixed

File: code/modules/halo/clothing/urf_commando.dm (hunk 11 above)
Comment (Eluxor) [reply code/modules/halo/clothing/urf_commando.dm]: fixed

File: code/modules/halo/clothing/urf_commando.dm (hunk 12 above)
Comment (Eluxor) [reply code/modules/halo/clothing/urf_commando.dm]: I just tried to fix it? I am not following you, I am not that well versed in code.

**Comment by bloxgate** [comment]
~~You need to make it an oxygen jetpack. Right now it's an empty jetpack. You
could only get rid of that `New()` if you changed it to be an oxygen
jetpack as well.~~

My bad. It seems like the diff I was looking at was outdated.

**Review by bloxgate — APPROVED** [review]

**Comment by BDpuffy420** [comment]
It’s was still failing Travis

File: code/modules/halo/structures/gunvendors.dm [hunk 14]
```diff
@@ -215,20 +217,26 @@
...
+	/obj/item/clothing/head/helmet/urfccommander/beretofficer = 7,
+	/obj/item/clothing/mask/gas/soebalaclava = 7,
+	/obj/item/clothing/mask/gas/soebalaclava/open = 7,
+	/obj/item/clothing/gloves/soegloves/urfc = 8,
 	/obj/item/clothing/shoes/magboots/urfc = 8,
-	/obj/item/weapon/armor_patch = 4)
+	/obj/item/weapon/armor_patch = 4,
+	/obj/item/clothing/head/helmet/soe = 6,
+	/obj/item/clothing/suit/armor/special/soe = 6,
+	/obj/item/weapon/tank/jetpack/urfc = 6 )
```
Comment (bloxgate) [inline code/modules/halo/structures/gunvendors.dm]: This doesn't exist.

**Review by bloxgate — DISMISSED** [review]

**Review by BDpuffy420 — APPROVED** [review]

**Pull request merged**